### PR TITLE
disttask/ddl: add retry for sessionExpired

### LINKED ISSUE #48546: add index failed with error “Error 1105 (HY000): mutex: session is expired”
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
1、run workload
2、add index for one table
3、inject pdleader io delay 1s 

case config：
      tag: "ha-test-add-index"
      workloads:
        - type: sysbench
          storage: s3://benchmark/sysbench_64_7000w
          db: sysbench_64_7000w
          tableNum: 64
          tableSize: 70000000
          threads: 64
          subType: oltp_read_write
          ignoreErrors: "all"
      cases:
        - name: ha_pd(leader)_io_delay(1s)
          faultType: iochaos_io_delay(volumePath:/var/lib/pd,path:/var/lib/pd/data/**/*,delay:1s,percent:100)
          selector: pd(leader)
          warmUpTime: 1m
          period: "@every 5m"
          faultDuration: 2m
          faultTotalRunTime: 30m
          ddl:
            addIndex:
              - db: "sysbench_64_7000w"
                table: "sbtest1"
                timeout: 60m
                times: 5
                index_col_name: "c"
                job_type: "ingest"
                admin_check: false
                count_check: false

### 2. What did you expect to see? (Required)
add index can success

### 3. What did you see instead (Required)
add index failed with error “Error 1105 (HY000): mutex: session is expired”

add index failed at 2023-11-12 14:14:05 (Error 1105 (HY000): mutex: session is expired)
operatorLogs:
[2023-11-12 14:10:07] ###### start adding index
alter table sbtest1 add index index_test_1699769407963 (c)
[2023-11-12 14:10:07] ###### wait for ddl job finish

### 4. What is your TiDB version? (Required)
git hash：700beafa79844b7b48dcba1c452ea3ff49d8f271

### PR DESCRIPTION
**Pull request by ywqzzy**
### What problem does this PR solve?

Issue Number: close #48546 

Problem Summary:
When trying to acquire dist lock on etcd, we might encounter sessionExpired error.
### What is changed and how it works?
1. Add retry backoffer to acquire lock.
### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  >   

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: br/pkg/lightning/common/retry.go [hunk 1]
```diff
@@ -79,15 +80,16 @@ var retryableErrorIDs = map[errors.ErrorID]struct{}{
 	ErrKVReadIndexNotReady.ID():   {},
 	ErrKVIngestFailed.ID():        {},
 	ErrKVRaftProposalDropped.ID(): {},
-	// litBackendCtxMgr.Register may return the error.
-	ErrCreatePDClient.ID(): {},
```
Comment (tangenta) [inline br/pkg/lightning/common/retry.go:83]: Why remove this?

Comment (ywqzzy) [reply br/pkg/lightning/common/retry.go:83]: The real error is ErrClientGetMember

File: pkg/ddl/index.go [hunk 2]
```diff
@@ -770,9 +770,9 @@ func pickBackfillType(ctx context.Context, job *model.Job, unique bool, d *ddlCt
 			if err != nil {
 				return model.ReorgTypeNone, err
 			}
-			if variable.EnableDistTask.Load() {
-				_, err = ingest.LitBackCtxMgr.Register(ctx, unique, job.ID, d.etcdCli, job.ReorgMeta.ResourceGroupName)
-			} else {
+
+			// for dist task, we Register BackCtx in backfillDistScheduler.Init.
+			if !variable.EnableDistTask.Load() {
```
Comment (tangenta) [inline pkg/ddl/index.go:775]: We can remove the `Register()` code here. Later I will file a PR to implement "only one task can run at the same time".

Comment (tangenta) [reply pkg/ddl/index.go:775]: #48645

File: pkg/ddl/ingest/backend.go [hunk 3]
```diff
@@ -192,13 +195,29 @@ func (bc *litBackendCtx) Flush(indexID int64, mode FlushMode) (flushed, imported
 		return true, false, nil
 	}
 
-	// Use distributed lock if run in distributed mode).
+	// Use distributed lock if run in distributed mode with local sort
+	// to avoid too many sst files for tikv to compact.
 	if bc.etcdClient != nil {
+		failpoint.Inject("mockSessionExpired", func() {
+			logutil.BgLogger().Info("ywq test trigger....")
```
Comment (ywqzzy) [inline pkg/ddl/ingest/backend.go:202]:
```suggestion
```

**Pull request closed without merging**